### PR TITLE
Bump `network-transport-tcp` & drop `network-transport-inmemory`

### PR DESCRIPTION
**Pull request by deepfire**
1. Bump `network-transport-tcp` for the https://github.com/input-output-hk/cardano-node/issues/296 fix.
2. Remove `network-transport-inmemory` dependency (by @avieth)

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -95,7 +95,7 @@ packages:
 # As above, seems relatively straightforward to PR.
 - location:
     git: https://github.com/avieth/network-transport-tcp
-    commit: 2634e5e32178bb0456d800d133f8664321daa2ef # csl-0.6.0
+    commit: 46f2942423d9ad3c81040565a9e9d5b9e08ddcc4 # csl-0.6.0
```
Comment (dcoutts) [inline stack.yaml:98]: What about the `cabal.project`?

Comment (dcoutts) [reply stack.yaml]: `cardano-repo-tool update-hash -r cardano-byron-proxy`

Comment (dcoutts) [reply stack.yaml:98]: Never mind, we're not keeping that in sync for the old things.

Comment (deepfire) [reply stack.yaml:98]: @dcoutts, fixed -- `46f2942423d9ad3c81040565a9e9d5b9e08ddcc4` is what should've been in `default.nix` as well.

**Comment by dcoutts** [comment]
I'd like to hold off merging this until it's clear we have all the release-essential things from @disassembler merged.

**Pull request closed without merging**